src/keyword_parser.test.js

```diff
@@ -0,0 +1,50 @@
+const KeywordParser = require('./keyword_parser');
+
+test('Does not consider a normal message to contain STOP keyword.', () => {
+  expect(KeywordParser.containsStopKeyword('Some normal message')).toBe(false);
+});
+
+test('Detects STOP keyword.', () => {
+  expect(KeywordParser.containsStopKeyword('STOP')).toBe(true);
+});
+
+test('Detects STOP keyword with spaces on either side.', () => {
+  expect(KeywordParser.containsStopKeyword('  STOP   ')).toBe(true);
+});
+
+test('Does not detect STOP keyword with spaces within the word.', () => {
+  expect(
+    KeywordParser.containsStopKeyword('VOTING IS TOP OF MIND FOR ME YAY')
+  ).toBe(false);
+});
+
+test('Detects STOP keyword with punctuation.', () => {
+  expect(KeywordParser.containsStopKeyword('STOP!')).toBe(true);
+});
+
+test('Detects STOP keyword in any upper/lower case.', () => {
+  expect(KeywordParser.containsStopKeyword('stop')).toBe(true);
+  expect(KeywordParser.containsStopKeyword('StOp')).toBe(true);
+});
+
+test('Detects STOP keyword alone in any upper/lower case even with punctuation.', () => {
+  expect(KeywordParser.containsStopKeyword('stop!!')).toBe(true);
+  expect(KeywordParser.containsStopKeyword('Stop!!')).toBe(true);
+  expect(KeywordParser.containsStopKeyword('S.T.O.P.')).toBe(true);
+});
+
+test('Does not detect lowercase stop in the context of a longer message if lowercase.', () => {
+  expect(
+    KeywordParser.containsStopKeyword(
+      "I'll stop by the polling place on the way to work"
+    )
+  ).toBe(false);
+});
+
+test('Detects STOP keyword in the context of longer message if in all uppercase.', () => {
+  expect(
+    KeywordParser.containsStopKeyword(
+      "That's what I am telling you, I have done what you told me to do to unsubscribe and YOU ARE STILL CONTACTING ME. STOP IT!!"
```

Comment:
hmm... do we have a sense of how common this is? I'm worried that some people just type in caps a lot and we're going to get false-positives. Maybe we try without this and see if we're still getting a lot of opt-outs going through?

Reply:
sounds good! should I split this PR up so that we can at least catch "STOP!!!!"